features/config/rest/impl/src/main/java/org/opennms/features/config/rest/impl/ConfigSwaggerConverter.java

```diff
@@ -0,0 +1,388 @@
+ * along with OpenNMS(R).  If not, see:
+ *      http://www.gnu.org/licenses/
+ *
+ * For more information contact:
+ *     OpenNMS(R) Licensing <license@opennms.org>
+ *     http://www.opennms.org/
+ *     http://www.opennms.com/
+ *******************************************************************************/
+
+package org.opennms.features.config.rest.impl;
+
+import com.fasterxml.jackson.annotation.JsonInclude;
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.SerializationFeature;
+import com.fasterxml.jackson.dataformat.yaml.YAMLFactory;
+import io.swagger.v3.oas.models.*;
+import io.swagger.v3.oas.models.info.Info;
+
+import io.swagger.v3.oas.models.media.*;
+import io.swagger.v3.oas.models.parameters.Parameter;
+import io.swagger.v3.oas.models.parameters.RequestBody;
+import io.swagger.v3.oas.models.responses.ApiResponse;
+import io.swagger.v3.oas.models.responses.ApiResponses;
+import org.opennms.features.config.dao.api.ConfigItem;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.math.BigDecimal;
+import java.util.*;
+import java.util.function.BiConsumer;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+public class ConfigSwaggerConverter {
+    private final Logger LOG = LoggerFactory.getLogger(ConfigSwaggerConverter.class);
+
+    private final Map<ConfigItem, Schema<?>> schemasByItem = new LinkedHashMap<>();
+    private final Map<ConfigItem, String> pathsByItem = new LinkedHashMap<>();
+    private final Map<String, PathItem> pathItemsByPath = new LinkedHashMap<>();
+
+    private final OpenAPI openAPI = new OpenAPI();
+
+    private String prefix = "/";
+
+    public String convertToString(ConfigItem item, String prefix, String acceptType) {
+        OpenAPI openapi = convert(item, prefix);
+
+        try {
+            return convertOpenAPIToString(openapi, acceptType);
+        } catch (JsonProcessingException e) {
+            return "";
+        }
+    }
+
+    /**
+     * convert open api object to specific string (default is yaml)
+     *
+     * @param openapi schema
+     * @param acceptType (json / yaml)
+     * @return
+     * @throws JsonProcessingException
+     */
+    public String convertOpenAPIToString(OpenAPI openapi, String acceptType) throws JsonProcessingException {
+        ObjectMapper objectMapper;
+        try {
+            org.springframework.http.MediaType mediaType = org.springframework.http.MediaType.valueOf(acceptType);
+            if (org.springframework.http.MediaType.APPLICATION_JSON.equals(mediaType) && mediaType != null) {
+                objectMapper = new ObjectMapper();
+            } else {
+                objectMapper = new ObjectMapper(new YAMLFactory());
+            }
+        } catch (Exception e) {
+            LOG.warn("UNKNOWN MideaType: " + acceptType + " error: " + e.getMessage());
```

Comment:
maybe explaining fallback in error message:
LOG.warn("UNKNOWN MideaType: " + acceptType + " error: " + e.getMessage()+ " using media type = yaml instead.");

Reply:
updated